otherlibs/stdune/src/io.mli

```diff
@@ -9,6 +9,10 @@ val input_lines : in_channel -> string list
     unrelated channels because it uses a statically-allocated global buffer. *)
 val copy_channels : in_channel -> out_channel -> unit
 
+(** This is raised when [read_all] or similar functions try to operate on files larger than [Sys.max_string_length]. This should be an issue only on 32-bit systems where this is 16 MiB. *)
+exception Large_file
+
+(** Read all of a channel. Can raise [Large_file]. *)
 val read_all : in_channel -> string
```

Comment:
Likewise, should probably be `read_all_exn`.

Reply:
Yes, even if in most cases we don't have to handle it, it might be a good nudge. I think I'll make that change in a separate PR.